app/models/mhv_account.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+require 'mhv_ac/client'
+class MhvAccount < ActiveRecord::Base
+  include AASM
+
+  # Everything except ineligible accounts should be able to transition to :needs_terms_acceptance
+  ALL_STATES = %i(unknown needs_terms_acceptance ineligible registered upgraded register_failed upgrade_failed).freeze
+  after_initialize :setup
```

Comment:
So the check for whether you have accepted the terms of service or not happens when the MhvAccount is initialized. See the setup step in the after_initialize callback. Technically it should never be the case that the account will be in "needs_terms_acceptance" if terms have been accepted.